src/app/shared/components/catalog/catalog.component.html

```diff
@@ -0,0 +1,27 @@
+<section class="catalog__cards">
+  <article *ngFor="let product of products" class="card" id="catalog-card-{{product.id}}">
+    <a class="card__item-link" routerLink="/product/{{product.id}}">
+      <div class="card__content">
+        <img alt="{{product.title}}" class="card__img" src="{{product.image}}" title="{{product.title}}">
+        <div class="card__details">
+          <p class="card__name">{{ product.title }}</p>
+          <app-rating [rating]="product.rating.rate"/>
+          <p class="card__price">{{ product.price + '$' }}</p>
```

Comment:
you can't be sure that you will receive all data
you can add ngIf or .? for properties

Reply:
please, fix it everywhere